[docs] Clarify how ILM affects template setup options

dedemorton opened the pull request:
Replaces elastic/beats#12419 plus adds a small edit

ycombinator commented on libbeat/docs/shared-template-load.asciidoc:
```diff
@@ -85,6 +85,8 @@ output.elasticsearch.index: "customname-%{[{beat_version_key}]}-%{+yyyy.MM.dd}"
 setup.template.name: "customname"
 setup.template.pattern: "customname-*"
 -----
+WARNING: If {ref}/index-lifecycle-management.html[Index Lifecycle Management] is enabled (which is the default), `setup.template.name` and `setup.template.pattern` are ignored.
```
Will this take users to `index-lifecycle-management.html`? From what I can tell from the online docs, the page URL is `ilm.html`: https://www.elastic.co/guide/en/beats/filebeat/master/ilm.html.

Also, should we make the same clarification over here as well? https://www.elastic.co/guide/en/beats/filebeat/master/ilm.html#setup-ilm-option

dedemorton replied:
TBH, I just fixed the link provided by the original contributor (didn't really add my own changes).  I think you're right that we need to send users to the Beats docs, so I'll fix that!

dedemorton commented:
@ycombinator Ok, I've added a warning to the ILM docs, too. BTW, I think we can't way that ILM is the default because it depends on the cluster. It's only the default if you are connecting to a cluster that supports ILM. That's clarified in the ILM docs. So I added the weasel word "typically".
